src/txdb.cpp

```diff
@@ -35,21 +33,25 @@ CCoinsViewDB::CCoinsViewDB(size_t nCacheSize, bool fMemory, bool fWipe) : db(Get
 }
 
 bool CCoinsViewDB::GetCoins(const uint256 &txid, CCoins &coins) const {
+    boost::shared_lock<boost::shared_mutex> lock(utxo);
```

Comment:
Why is that needed ? Isn't the db ACID already ? That'd be worrying if it wasn't.